lib/app.js

```diff
@@ -404,17 +405,20 @@ module.exports = {
             return value;
           }
         }
-      }];
-
-      getPrompt(argv, schema, function(results) {
-        if (path.relative('.', results.public).match(/^\./)) {
-          console.log(chalk.red('init cancelled - the public directory must be within the current working directory'));
-          process.exit(1);
+      };
+      schemaPublic.conform = function(value) {
+        if (path.relative('.', value).match(/^\./)) {
+          schemaPublic.message = 'Please use a sub directory of current working directory';
+          return false;
         }
-        if (!fs.existsSync(results.public)) {
-          console.log(chalk.red('init cancelled - the directory you entered does not exist'));
-          process.exit(1);
+        if (!fs.existsSync(value)) {
+          schemaPublic.message = 'Please use an existing directory';
```

Comment:
The current restrictions work, but perhaps could be relaxed to facilitate some more advanced workflows. Lets leave it as is for now, but consider changing this in a future release